src/charts/heatmap/heatmap.directive.js

```diff
@@ -271,8 +286,20 @@ angular.module('patternfly.charts').directive('pfHeatmap', function ($compile, $
 
         //Adding events
         blocks.on('mouseover', function () {
+          var fillColor;
           blocks.call(highlightBlock, false);
-          d3.select(this).call(highlightBlock, true);
+          if (scope.maxSize < 16) {
```

Comment:
But scope.maxSize is passed in, it does not mean that is the size of the blocks. For example, an application may  want a maximum block size of 100. That means that even if there are only 2 data items, only use 100x100px blocks, do not fill the entire area with huge blocks. We set the max block size small in these examples to show small blocks. 

If I set a maxSize of 100 but there are 10,000,000 blocks, the computed block size will be small and THAT is when we want to use this functionality.